File: pset2/readability/readability.c
#include <cs50.h>
#include <stdio.h>
#include <ctype.h>
#include <string.h>
#include <math.h>

int count_letters(string input);
int count_words(string input);
int count_sent(string input);

int main(void)
{
    string text = get_string("Text: ");

    int letters = count_letters(text), words = count_words(text), sent = count_sent(text);

    // Avg. number of letters/100 words
    float L = (float)(letters * 100) / words;

    // Avg. number of sentences/100 words
    float S = (float)(sent * 100) / words;

    // Compute the Coleman-Liau index
    int index = round(0.0588 * L - 0.296 * S - 15.8);

    // Output the reading level
    if (index < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (index > 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Grade %i\n", index);
    }
}

// Counts the number of letters in input
int count_letters(string input)
{
    int count = 0;

    for (int i = 0, n = strlen(input); i < n; i++)
    {
        if (isalpha(input[i]))
        {
            count++;
        }
    }

    return count;
}

// Counts the number of words in input
int count_words(string input)
{
    int count = 0;

    for (int i = 0, n = strlen(input); i < n; i++)
    {
        if (isspace(input[i]))
        {
            count++;
        }
    }

    //add last word
    count++;

    return count;
}

// Counts the number of sentences in input
int count_sent(string input)
{
    int count = 0;

    for (int i = 0, n = strlen(input); i < n; i++)
    {
        char c = input[i];

        if (c == 33 || c == 46 || c == 63)
        {
            count++;
        }
    }

    return count;
}

File: pset6/readability/readability.py
from cs50 import get_string


def main():
    text = get_string("Text: ")

    # Count number of letters, words, and sentences
    letters = count_letters(text)
    words = count_words(text)
    sent = count_sent(text)

    # Avg number of letters/100 words
    L = letters * 100 / words

    # Avg number of sentences/100 words
    S = sent * 100 / words

    # Compute the Coleman-Liau index
    index = round(0.0588 * L - 0.296 * S - 15.8)

    # Output the reading level
    if index < 1:
        print("Before Grade 1")
    elif index > 16:
        print("Grade 16+")
    else:
        print("Grade", index)


# Count number of letters in input


def count_letters(txt):
    count = 0
    for char in txt:
        if char.isalpha():
            count += 1
    return count

# Count number of words in input


def count_words(txt):
    count = 0

    for char in txt:
        if char.isspace():
            count += 1
    # Add last word
    count += 1

    return count

# Count number of sentences in input


def count_sent(txt):
    count = 0

    for char in txt:
        if char == "." or char == "!" or char == "?":
            count += 1

    return count


main()
